eng/build.yml

```diff
@@ -88,26 +88,59 @@ jobs:
           BlobFeedUrl: $(PB_PublishBlobFeedUrl)
           PublishType: $(_PublishType)
 
-    - ${{ if eq(parameters.agentOs, 'Windows_Performance') }}:
-      - script: eng\common\CIBuild.cmd 
+    - ${{ if eq(parameters.agentOs, 'Windows_Perf_CI') }}:
+      - script: build.cmd
                   -configuration $(_BuildConfig)
+                  -ci -msbuildEngine dotnet
                   -performanceTest
-                  -msbuildEngine dotnet 
-                  $(_PerformanceArgs)
         displayName: Build
         env:
           BuildConfig: $(_BuildConfig)
 
-    - ${{ if eq(parameters.agentOs, 'Linux_Performance') }}:
-      - script: eng/common/cibuild.sh 
+    - ${{ if eq(parameters.agentOs, 'Linux_Perf_CI') }}:
+      - script: ./build.sh
                   --configuration $(_BuildConfig)
-                  --performancetest
-                  $(_PerformanceArgs)
+                  --ci --performancetest
         displayName: Build
         env:
           BuildConfig: $(_BuildConfig)
 
-    - ${{ if notIn(parameters.agentOs, 'Windows_NT', 'Windows_NT_FullFramework', 'Windows_NT_TestAsTools', 'Windows_Performance', 'Linux_Performance') }}:
+    - ${{ if eq(parameters.agentOs, 'Windows_Perf_Helix') }}:
+        - checkout: self
+          clean: true
+        - script: $(Build.SourcesDirectory)\build.cmd -configuration $(_BuildConfig) -ci -msbuildEngine dotnet
+          displayName: Pre-Build
+        - script: (robocopy $(Build.SourcesDirectory) $(Build.SourcesDirectory)\helix-package /NP /E /XD $(Build.SourcesDirectory)\helix-package $(Build.SourcesDirectory)\.dotnet $(Build.SourcesDirectory)\artifacts\bin\Tests $(Build.SourcesDirectory)\artifacts\obj\$(_BuildConfig)\Sdks\Microsoft.NET.Build.Extensions $(Build.SourcesDirectory)\artifacts\bin\$(_BuildConfig)\Sdks\Microsoft.NET.Build.Extensions) ^& IF %ERRORLEVEL% LEQ 1 exit 0
+          displayName: Selection of sources
+        - template: /eng/common/templates/steps/send-to-helix.yml
+          parameters:
+            HelixSource: 'pr/dotnet/sdk/$(Build.SourceBranch)'
+            HelixType: 'test/sdk_$(_BuildConfig)/'
+            HelixAccessToken: $(HelixApiAccessToken)
+            HelixTargetQueues: 'Windows.10.Amd64.ClientRS1.Perf' # using a dedicated private Helix queue (perfsnakes)
+            HelixPreCommands: 'set "BV_UPLOAD_SAS_TOKEN=$(BenchViewUploadToken)"'
+
+            WorkItemTimeout: 4:00
```

Comment:
How long does your perf testing run? Does this need to be 4 hours? Can it be shorter?

Reply:
About 1 hour 15 minutes - I'll shorten it up.